Make the exception raised from "numpy.dtype(numpy.void, (INT,))" less cryptic

chandlerzuo opened the pull request:
Summary:
https://fb.facebook.com/groups/582508038765902/permalink/736710343345670/?comment_id=824042307945806&reply_comment_id=824318864584817

numpy.dtype(numpy.void, (<INT>, )) raises a cryptic message "invalid itemsize in generic type tuple" that is hard to debug.

This diff adds the message to ask the user to investigate the error causing blob.

Differential Revision: D13973359

ezyang commented on caffe2/python/schema.py:
```diff
@@ -731,6 +731,14 @@ def set(self, dtype=None, blob=None, metadata=None, unsafe=False):
+                    "If this blob is the output of some operation, "
+                    "please verify the input of that operation has proper type, "
+                    "especially if it is created with "
+                    "unsafe_shape_infererenc().".format(dtype, blob)
```
nit: spelling inference

chandlerzuo commented:
@ezyang Would you mind taking another look? Also, do you think the CI failures are false here?

chandlerzuo committed:
Make the exception raised from "numpy.dtype(numpy.void, (INT,))" less cryptic (#16809)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/16809

https://fb.facebook.com/groups/582508038765902/permalink/736710343345670/?comment_id=824042307945806&reply_comment_id=824318864584817

numpy.dtype(numpy.void, (<INT>, )) raises a cryptic message "invalid itemsize in generic type tuple" that is hard to debug.

This diff adds the message to ask the user to investigate the error causing blob.

Reviewed By: kennyhorror

Differential Revision: D13973359

fbshipit-source-id: c84149b053e76be78efe83684a9e2d960c562450
